Small improvements/Upgrade guide

emptynick opened the pull request:
Added the upgrade-guide for 1.2:
- New configurations
- Asset-loading (remove old assets)
- User-BREAD controller

Fixed some small things:
- Simplified edit/add condition in bread-view. See https://github.com/the-control-group/voyager/pull/3856#discussion_r244538249
- Radio buttons didn't use the display-text, but the key instead. Fixes #3907 
- Use `empty()` instead of `count()` for multiple-select. Fixes #3879 
- Replaced hardcoded "BREAD". Fixes #3548 
- Lost order-params when searching and lost search params when ordering. Fixes #3923
- Added `default` prop to number formfield. Fixes #3503
- Renamed `Model namespace` to `Fully qualified Model name`. Closes #1908

Enhancements:
- Added customizable formfield-views. Fixes #3301 Closes #3738
- Added documentation for custom actions
- Action specific validation rules. Fixes #2034
- Added BREAD-scopes. Fixes #3941 
- Added advanced installation docs. Fixes #3719 
- Replaced user-models settings cast with an accessor/mutator-pair. Closes #3950

emptynick committed:
Simplify edit/add conditions

emptynick committed:
Merge branch '1.x' into improvements

emptynick committed:
RadioButton - Get value from options

Fixes #3907

emptynick committed:
Use empty instead of count

Fixes #3879

emptynick committed:
Upgrade guide

emptynick committed:
Add rest of media-config to upgrade-guide

emptynick committed:
Add missing image

fletch3555 commented on resources/views/bread/edit-add.blade.php:
```diff
@@ -1,15 +1,20 @@
+@php
+    $edit = !is_null($dataTypeContent->getKey());
+    $add  = is_null($dataTypeContent->getKey());
```
Do you need the `$add`? I don't see it used anywhere

emptynick replied:
Not really. Just added it for consistency.

emptynick committed:
Merge remote-tracking branch 'upstream/1.x' into improvements

emptynick committed:
Empty instead of count in read-view

emptynick committed:
Remove harcoded BREAD

#3548

emptynick committed:
Change string to boolean

https://github.com/the-control-group/voyager/blob/1.x/resources/views/formfields/checkbox.blade.php#L6

emptynick committed:
Customize formfield views

emptynick committed:
BREAD scoping

emptynick committed:
Use scope for other BREAD actions

emptynick committed:
Don't show query-string in breadcrumbs

emptynick committed:
Document scoping

emptynick committed:
Dont remove order or search from url-query when doing the other action

When searching, the ordering was lost, when ordering, the search was lost.

emptynick committed:
StyleCI

emptynick committed:
Add default-prop to number formfield

emptynick committed:
Renamed model-namespace to FQN

emptynick committed:
Document custom actions

emptynick committed:
Action specific validation rules/docs-fix

emptynick committed:
Improve condition

emptynick committed:
Remove leftover methods

From old can/canOrFail methods

emptynick committed:
Require Hooks 1.2.0

emptynick committed:
Remove RTL config from login

emptynick committed:
Require Hooks > 1.2.1

emptynick committed:
Document #3867

emptynick committed:
Merge branch '1.x' into improvements

emptynick committed:
Add advanced installation

emptynick committed:
Replace settings-cast with accessor/mutator

emptynick committed:
Merge branch '1.x' into improvements

emptynick committed:
Return empty array

emptynick committed:
Merge branch 'improvements' of https://github.com/emptynick/voyager into improvements

emptynick committed:
Default to app-locale

emptynick committed:
Simplify usage

fletch3555 committed:
Merge branch '1.x' into improvements
